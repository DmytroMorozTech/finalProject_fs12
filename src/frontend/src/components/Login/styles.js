import { makeStyles } from '@material-ui/core/styles'

export default makeStyles((theme) => ({

  login: {
    width: '100vw',
    height: '100vh',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.palette.background.default,
    position: 'absolute',
    top: 0,
    left: 0,
    zIndex: 150
  },

  loginCard: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center'
  },

  signUpLineWrapper: {
    zIndex: 150,
    display: 'flex',
    justifyContent: 'center',
    fontSize: theme.typography.sh3.fontSize
  },

  signUpLineLink: {
    fontSize: theme.typography.sh3.fontSize,
    textDecoration: 'none',
    '& :hover': {
      textDecoration: 'underline'
    }
  },

  signInLogo: {
    position: 'absolute',
    top: '2rem',
    left: '3rem',
    zIndex: 150
  }
}))
