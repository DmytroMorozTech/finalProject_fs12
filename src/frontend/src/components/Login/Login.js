import LoginCard from './loginCard/LoginCard'
import {useSelector} from 'react-redux'
import {isLoadingUserSelector} from '../../redux/User/userSelector'
import Preloader from '../../shared/Preloader/Preloader'
import React from 'react'
import LinkedinLogo from '../../shared/LinkedinLogo/LinkedinLogo'
import styles from './styles'
import {Link} from 'react-router-dom'

const Login = () => {
  const isLoadingUser = useSelector(isLoadingUserSelector)
  const classes = styles()
  return isLoadingUser ? <Preloader/> : (
    <div className={classes.login}>
      <div className={classes.signInLogo}>
        <LinkedinLogo/>
      </div>
      <div className={classes.loginCard}>
        <LoginCard/>
      </div>
      <div className={classes.signUpLineWrapper}>
        <p>New to LinkedIn?</p>
        <Link to="/signup" className={classes.signUpLineLink}>Join now</Link>
      </div>
    </div>
  )
}

export default Login
