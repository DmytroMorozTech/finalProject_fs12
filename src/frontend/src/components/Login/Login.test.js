import { shallow } from 'enzyme'
import * as redux from 'react-redux'
import React from 'react'
import Login from './Login'

jest.mock('./styles', () => () => ({}))

describe('should render Login component', () => {
  let spyOnUseSelector

  let component

  const setUp = () => shallow(<Login />)

  beforeEach(() => {
    spyOnUseSelector = jest.spyOn(redux, 'useSelector')
    spyOnUseSelector.mockReturnValue(false)

    component = setUp()
  })
  afterEach(() => {
    jest.restoreAllMocks()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})

describe('should render Login component', () => {
  let spyOnUseSelector

  let component

  beforeEach(() => {
    spyOnUseSelector = jest.spyOn(redux, 'useSelector')
    spyOnUseSelector.mockReturnValue(true)

    component = shallow(<Login />)
  })
  afterEach(() => {
    jest.restoreAllMocks()
  })
  it('should render Preloader', () => {
    let wrapper = component.find('Preloader')
    expect(wrapper.length).toBe(1)
  })
})
