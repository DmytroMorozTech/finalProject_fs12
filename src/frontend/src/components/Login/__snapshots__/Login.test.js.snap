// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login component to match snapshot 1`] = `
<div>
  <div>
    <LinkedinLogo />
  </div>
  <div>
    <LoginCard />
  </div>
  <div>
    <p>
      New to LinkedIn?
    </p>
    <Link
      to="/signup"
    >
      Join now
    </Link>
  </div>
</div>
`;
